### PR TITLE
Move and rename map node ID type

### PR DESCRIPTION
**Pull request by pavelkalinnikov**
The node ID type logically belongs to the `merkle` subfolder which holds the algorithmic
aspect of Merkle trees. This change moves the `NodeID2` type, which is now known simply
as `node.ID`. It also untangles some dependencies that blocked this change.

#1855
#2378

### CODE REVIEW

**Comment by pavelkalinnikov** [comment]
Review commit-by-commit might be convenient.

**Comment by pavelkalinnikov** [comment]
It's better to review #2526 first. This PR will become much simpler after rebasing.

**Comment by pavelkalinnikov** [comment]
@pphaneuf This is ready for review now. It's a big but simple one: in a nutshell, just a move + rename + update usages.

**Comment by pavelkalinnikov** [comment]
@mhutchinson PTAL too, it affects `batchmap`.

File: merkle/coniks/coniks.go [hunk 1]
```diff
@@ -107,27 +107,27 @@ func (m *Hasher) BitLen() int {
...
-// writeMaskedNodeID writes the node ID bits to the buffer, padded with zero
-// bits to the byte Size() of the hashes in use by this hasher.
+// writeMaskedID writes the node ID bits to the buffer, padded with zero bits
+// to the byte Size() of the hashes in use by this hasher.
 //
 // TODO(pavelkalinnikov): We must not use BitLen() and Size() interchangeably.
 // The tree height and hash size could be different.
 // TODO(pavelkalinnikov): Padding with zeroes doesn't buy us anything, as the
 // depth is also written to the Buffer.
-func (m *Hasher) writeMaskedNodeID(b *bytes.Buffer, node tree.NodeID2) {
```
Comment (mhutchinson) [inline merkle/coniks/coniks.go:117]: opinion: I preferred the previous method name. In normal conversation one would still refered to `node.ID` as a "node ID".

**Review by mhutchinson — APPROVED** [review]

File: merkle/coniks/coniks.go (hunk 1 above)
Comment (pavelkalinnikov) [reply merkle/coniks/coniks.go:117]: Fair point, restored the original name.

**Pull request merged**